lightning/src/ln/msgs.rs

```diff
@@ -896,6 +896,9 @@ pub trait ChannelMessageHandler : MessageSendEventsProvider {
 	// Error:
 	/// Handle an incoming error message from the given peer.
 	fn handle_error(&self, their_node_id: &PublicKey, msg: &ErrorMessage);
+
+	/// Advertise onion message forwarding support in broadcasted node announcements.
+	fn advertise_onion_message_support(&self);
```

Comment:
I get that we're trying to only advertise the feature if we have a handler that will support it, but this seems like it should be more of a configuration parameter. Otherwise, we're shoehorning it into a trait meant for handling messages even though this only affects message broadcasting.

Trying to think if there is a better way of accomplishing this...

Reply:
Some questions / ideas to bounce of you both:

Do we expect the user to ever need to write their own `OnionMessageHandler`? Or will they always configure `OnionMessenger` with individual handlers for the different types of onion message payloads? If the latter, instead of using `IgnoringMessageHandler `, maybe we could have it be `Option<OnionMessenger>` and use its presence to signal if onion message forwarding is supported? Might be worth doing this regardless instead of having two levels of configuration.

For `broadcast_node_announcement`, could this also take `InitFeatures` as a parameter? I guess we'd want only `PeerManager` to determine which features to pass to it. So could have a `broadcast_node_announcement` method on `PeerManager` (without the parameter), which delegates to `ChannelManager` (with the parameter). Though that means having a somewhat unrelated `broadcast_node_announcement` on `ChannelMessageHandler` or `MessageSendEventsProvider`. Alternatively, have `BackgroundProcessor` be responsible for calling `broadcast_node_announcement` on `ChannelManager` with features obtained from `PeerManager`.

Reply:
> I don't think so, there's plenty of things the ChannelManager is responsible for and needs to select the features of :(.

Not sure I understand here. Looks like `broadcast_node_announcement` is the only place it needs our features.

Reply:
Hmmm... I see . Though that would also mean the features advertised by `Init` may not be accurate if a custom channel message handler were used? Seems each handler may want to declare which features they support (i.e., implement), and the `PeerManager` would want to take the intersection of these for what is actually used in either `Init` or `ChannelAnnouncement`?

Reply:
er... s/intersection/union